### ISSUE TITLE
effect of scaling_w when training Hidden model

### ISSUE THREAD

**Issue by jackyhans1** [facebookresearch/stable_signature#42]
Hi!

Thank you for your hard work and constant contribution.

I've got a question about the effect of hyper-parameter  `scaling_w` when training Hidden model.

I've seen that you've provided default `scaling_w` to be 0.3 as mentioned in hidden/README.md file.

I want to ask if you've tried other numbers for `scaling_w` except for 0.3. And if so, can I get to know the exact effect of changing `scaling_w`?

As I tried 1.0 for `scaling_w`, the overall bit_accuracy for train / val increased, the overfitting occurred. 

The following is the command I used.

`torchrun --nproc_per_node=1 main.py --dist False   --val_dir /mnt/new_vol/coco/val2014/ --train_dir /mnt/new_vol/coco/train2014/ --output_dir output --eval_freq 50   --img_size 256 --num_bits 32  --batch_size 16 --epochs 1000   --scheduler CosineLRScheduler,lr_min=1e-6,t_initial=1000,warmup_lr_init=1e-6,warmup_t=5  --optimizer Lamb,lr=0.005   --p_color_jitter 0.0 --p_blur 0.0 --p_rot 1.0 --p_crop 1.0 --p_res 0.0 --p_jpeg 1.0   --scaling_w 1.0 --scale_channels False --attenuation none   --loss_w_type bce --loss_margin 1`

Does increasing the value of `scaling_w` cause overfitting?

The following is the log of epoch 250. This kind of overfitting happens consistently.

`{"train_loss_w": 0.22593628048896788, "train_loss_i": 0.7765025331497193, "train_loss": 0.22593628048896788, "train_psnr_avg": 14.022187692260742, "train_lr": 0.004267913399575829, "train_bit_acc_avg": 0.871840625, "train_word_acc_avg": 0.0068, "train_norm_avg": 59.161339520263674, "epoch": 250, "val_loss_w": 2.2488539677399855, "val_loss_i": 0.7039491923955771, "val_loss": 2.2488539677399855, "val_psnr_avg": 14.460047538463886, "val_bit_acc_avg": 0.7614808693910257, "val_word_acc_avg": 0.0, "val_norm_avg": 225.06417044615134, "val_bit_acc_none": 0.7614808693910257, "val_bit_acc_crop_01": 0.7515274439102564, "val_bit_acc_crop_05": 0.7610802283653846, "val_bit_acc_resize_05": 0.5568096454326923, "val_bit_acc_rot_25": 0.7057041266025641, "val_bit_acc_rot_90": 0.5079690004006411, "val_bit_acc_blur": 0.5004444611378205, "val_bit_acc_jpeg_50": 0.5323329827724359}`

I tried increasing the value of `scaling_w` to get higher bit accuracy because I'm suffering due to bit accuracy < 80% as I changed `num_bits 32` and image rotation distortion as the following.

```
if p_rot > 0:
            aff1 = K.RandomAffine(degrees=(-45,45), p=1)
            aff2 = K.RandomAffine(degrees=(30,30), p=1)
            aff3 = K.RandomAffine(degrees=(-30,-30), p=1)
            augmentations += [aff1]
            augmentations += [aff2, aff3]
```

Thanks again for your hard work.

**Comment by pierrefdz** [comment]
Hi!
There's a discussion about the influence of scaling_w over the bit accuracy of the watermarked generator in Appendix B.4 in the paper. For your specific case, yes maybe bigger scaling_w leads to overfitting.
If you want to train a model specifically for distillation into generative models, you can try to train with the same code but sweep with different hyperparameters as the one of the paper is hard to reproduce and very sensitive to hyperparams.
If you just want to train watermarking models, you can look at https://github.com/facebookresearch/videoseal instead

PS: and sorry, i hadnt seen the issue before...
